core/src/main/java/org/elasticsearch/search/fetch/subphase/ParentFieldSubFetchPhase.java

```diff
@@ -46,6 +46,11 @@ public void hitExecute(SearchContext context, HitContext hitContext) {
             return;
         }
 
+        if (hitContext.hit().getNestedIdentity() != null) {
+            // skip nested documents as they nested have a join field
```

Comment:
> put a parent/child inner hit under a nested inner hit

Right, that doesn't work. One can only nest a nested inner hit under a p/c inner hit, which is what this check is checking for.

>  Maybe we should instead fix getParentId to return null when a document does not have a _parent field using LeafReader.getDocsWithField?

+1 That is a cleaner solution.